### DISCUSSION TITLE
20240523 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#95] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x
- Borja (@BorjaPractica) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Index

Configuration overhaul. Applying the same changes I made in the Tracker:

- [x] https://github.com/torrust/torrust-index/issues/587
- [x] https://github.com/torrust/torrust-index/issues/589
- [x] https://github.com/torrust/torrust-index/issues/581
- [x] https://github.com/torrust/torrust-index/issues/596
- [ ] https://github.com/torrust/torrust-index/issues/599 (WIP)

WIP: https://github.com/torrust/torrust-index/issues/586

I was also adding TSL support to the Index:

- [x] https://github.com/torrust/torrust-index/issues/426

After enabling TSL, I realized the patch I was using to add timeouts does not work with TSL:

See https://github.com/torrust/torrust-index/issues/204#issuecomment-2115529214.

That's a problem in both the Tracker and Index (where we use Axum). I've opened a new [issue on the Axum repo](https://github.com/tokio-rs/axum/issues/2741).

### Torrust Website

PR reviews:

- https://github.com/torrust/torrust-website/pull/178
- https://github.com/torrust/torrust-website/pull/180

### Courses

I'm doing an online course this week 8 hours ([Tech Lead](https://engineeringmanager.academy/))

**Comment by grmbyrn** [comment]
## Torrust Website
### Opened Issues:
* [Add link to previous and next posts under each blog post](https://github.com/torrust/torrust-website/issues/179)
* [Improve Lighthouse score](https://github.com/torrust/torrust-website/issues/181)
* [Search bar](https://github.com/torrust/torrust-website/issues/182)

### Changes Committed:
* [Page for tags in blog post](https://github.com/torrust/torrust-website/pull/178) - completed requested changes so that `tagId` uses SvelteKit option of `$page.url.pathname` instead of `window`.
* [Make all tags links](https://github.com/torrust/torrust-website/pull/178) - moved the link for tags from `+layout.svelte` in `(blog-article)` route to `Tag` component.
* [Button for ShareUrl](https://github.com/torrust/torrust-website/pull/178) - completed requested changes so that `ShareUrl.svelte` uses SvelteKit to get URL.
* [Add link to previous and next posts under each blog post](https://github.com/torrust/torrust-website/pull/180) - Add new component called `PrevNextPost.svelte`. This will be placed below each blog post and will display the previous and next blog post.

### Work in Progress:
* [Search bar](https://github.com/torrust/torrust-website/issues/182) - two options for a search bar.

**Comment by mario-nt** [comment]
### GUI

- https://github.com/torrust/torrust-index-gui/pull/559
- https://github.com/torrust/torrust-index-gui/pull/558
- https://github.com/torrust/torrust-index-gui/pull/515


### Index
	
- https://github.com/torrust/torrust-index/issues/448
- https://github.com/torrust/torrust-index/issues/559
- https://github.com/torrust/torrust-tracker/issues/670


### PR reviews

- https://github.com/torrust/torrust-index-gui/pull/560#pullrequestreview-2064469293

### Training/Research


- Cerbos 101 research 
- Debugging with Rust analyzer
- Rust programming language book
- Tracker repo 101
- Rust (From and into, traits)
- UDP tracker protocol 101

### Other

- One to one with jose on authorization solution
